doc/walkthrough.rst

```diff
@@ -0,0 +1,139 @@
+workloads, such as database services, web applications, or a mixture. However, the underlying hosts themselves are
+configured to be OpenShift nodes and are configured identical to each other.
+
+To create a brand new cluster:
+
+.. note::
+
+    If you did not create a ContainerManager you can omit ``--container-manager``.
+
+.. include:: examples/commctl_create_cluster.rst
+
+Adding Hosts To The Cluster
+---------------------------
+
+Adding new hosts to Commissaire comes in two forms. Automatic registration and manual additions.
+
+Automatic Registration
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+    This will be streamlined next release.
+
+This is the preferred way to deal with new hosts. This method allows Commissaire and, if configured, the
+ContainerManager, to be ready to use the host.
+
+Before we start you will need `make-mime.py`_ script from cloud-init's ``tools`` directory.
+
+First, you will need to create a configuration file for ``cloud-config``. Generally you'll do this once
+and then change the ``cluster`` key for different clusters. Here is an example:
+
+.. literalinclude:: ../tools/cloud-init/commissaire.txt
+   :language: shell
+
+Next, you will need a ``cloud-config`` text file. Unless you have a custom one created already
+the following will work:
+
+.. literalinclude:: ../tools/cloud-init/cloud-config.txt
+   :language: yaml
+
+Now copy ``tools/cloud-init/part-handler.py`` into the same directory as``commissaire.txt`` and ``cloud-config.txt``.
+
+Then, use `make-mime.py`_ to put it all together into a ``user-data`` file we can use in cloud providers.
+
+.. _make-mime.py: http://bazaar.launchpad.net/~cloud-init-dev/cloud-init/trunk/view/head:/tools/make-mime.py
+
+.. note::
+
+    You may see "text/x-commissaire-host' for attachment 3 may be incorrect!". Don't worry! This is OK.
+
+.. code-block:: shell
+
+   $ python make-mime.py \
+            --attach cloud-config.txt:cloud-config \
+            --attach part-handler.py:part-handler \
+            --attach commissaire.txt:x-commissaire-host \
+            > user-data
+
+Now provide the new ``user-data`` file when provisioning new hosts in your cloud provider. When the new host starts
+it will automatically register into Commissaire.
+
+Manual Registration
+~~~~~~~~~~~~~~~~~~~
+You can also add hosts into Commissaire in a manual fashion. To do this you will need:
+
+1. The host to have ``sshd`` running
+2. The host to have ``sshd`` port open.
+3. The private key to an administrative user on the host (EG: ``root``)
+
+
+.. note::
+
+   If you want to create a new key jump down to Optional: Creating Keys
+
+
+Let's say you have a host called ``192.168.152.110`` which you'd like to add to the cluster ``my_cluster``. You also
+have a private key of the remote root user for ``192.168.152.110`` at ``/path/to/remote/hosts/priv/ssh_key``. The
+following command would add the host to the cluster:
+
+.. include:: examples/commctl_host_create.rst
```

Comment:
commctl_host_create.rst was updated to

> $ commctl host create --cluster my_cluster 192.168.152.110 */path/to/remote/hosts/priv/ssh_key*

but I'm still confused here. The list of required items still says I need the private key of an admin (ex root) account.

But... "You have a private key of the remote root user for ``192.168.152.110`` at ``/path/to/remote/hosts/priv/ssh_key``

The file path suggests that is the private key for the **host** `192.168.152.110` but the sentence says it's the private key of the remote root **user**.

Please clarify. I'm pretty sure it's all supposed to be talking about remote hosts keys, but I am not sure.